notebooks/corridor/corridor_bucharest_river-valley.qmd

```diff
@@ -0,0 +1,299 @@
+
+We use the highway geometries to create the street network using `sfnetworks`:
+
+```{r}
+network <- highways |>
+    as_sfnetwork(directed = FALSE)
+```
+
+We perform standard cleaning tasks on the network:
+
+```{r}
+# Simplify the graph, removing loops and double-edge connections
+# https://luukvdmeer.github.io/sfnetworks/articles/sfn02_preprocess_clean.html#simplify-network
+simplify <- function(net){
+    net|>
+        activate("edges") |>
+        # reorder the edges so that the shortest is kept
+        arrange(edge_length()) |>
+        filter(!edge_is_multiple()) |>
+        filter(!edge_is_loop())
+}
+
+clean <- function(net){
+    net |>
+        # subdivide edges by adding missing nodes
+        # https://luukvdmeer.github.io/sfnetworks/articles/sfn02_preprocess_clean.html#subdivide-edges
+        convert(to_spatial_subdivision, .clean = TRUE) |>
+        # remove pseudo-nodes
+        # https://luukvdmeer.github.io/sfnetworks/articles/sfn02_preprocess_clean.html#smooth-pseudo-nodes
+        convert(to_spatial_smooth, .clean = TRUE) |>
+        # run simplification steps
+        simplify() |>
+        # keep only the main connected component of the network
+        activate("nodes") |>
+        filter(group_components() == 1)
+}
+
+network_cleaned <- clean(network)
+```
+
+Visualize cleaned network:
+
+```{r}
+leaflet() |>
+    addTiles() |>
+    addPolylines(data = network_cleaned |> get_edges() |> get_geom_latlon(), color = "black") |>
+    addCircles(data = network_cleaned |> get_nodes() |> get_geom_latlon(), color = "red")
+```
+
+
+## 3. Corridor edge delineation ----
+
+We carry out the delineation of the river corridor in few steps.
+
+First, we consider the nodes of the network within the area of interest (here defined using the bounding box of the city) and label them according to which side of the river they fall on:
+
+```{r}
+# split polygon in two fragments using the waterway geometry as dividing line
+# if more than two fragments are obtained, keep only the two largest fragments
+split_river_sides <- function(polygon, waterway){
+    # split polygons into fragments using waterway
+    fragments <- st_as_sf(polygon) |>
+        st_split(st_geometry(waterway)) |>
+        st_collection_extract()
+
+    # sort fragments according to area
+    fragments_sorted <- fragments |>
+        mutate(area = st_area(fragments)) |>
+        arrange(-area)  # descending order
+
+    # pick the two largest fragments
+    st_geometry(fragments_sorted[1:2,])
+}
+```
+
+```{r}
+# determine area of interest: city bounding box
+bbox_polygon <- city_boundary |>
+    st_bbox() |>
+    st_as_sfc()
+
+# split the area of interest in the two river sides
+river_sides <- split_river_sides(bbox_polygon, waterway)
+
+# label nodes with side of the river they fall on
+network_classified <- network_cleaned |>
+    activate("nodes") |>
+    mutate(side_1 = node_intersects(river_sides[1,])) |>
+    mutate(side_2 = node_intersects(river_sides[2,]))
```

Comment:
If `side_1` and `side_2` are exclusive, then this should work too:
```suggestion
    mutate(side_1 = node_intersects(river_sides[1,]),
           side_2 = !side_1)
```
The advantage is that you do not run `node_interesects()` again. You can also put both in the same `mutate()` call.

Reply:
This does not work as the bbox considered for the nodes is different from the extended bbox used for the edges.